src/main_controller.ts

```diff
@@ -368,4 +370,67 @@ export class MainController implements vscode.Disposable {
             return;
         }
     }
+
+    private async restartExtensionHostPrompt(message: string): Promise<void> {
+        vscode.window.showInformationMessage(message, "Restart").then((value) => {
+            if (value == "Restart") {
+                this.logger.debug("Restarting extension host");
+                vscode.commands.executeCommand("workbench.action.restartExtensionHost");
+            }
+        });
+    }
+
+    private async verifyExperimentalAffinity(): Promise<void> {
+        const extensionsConfiguration = vscode.workspace.getConfiguration("extensions");
+        const affinityConfiguration = extensionsConfiguration.inspect<{ [key: string]: [number] }>(
+            "experimental.affinity",
+        );
+
+        const affinityConfigWorkspaceValue = affinityConfiguration?.workspaceValue;
+        if (affinityConfigWorkspaceValue && EXT_ID in affinityConfigWorkspaceValue) {
+            this.logger.debug(
+                `Extension affinity value ${affinityConfigWorkspaceValue[EXT_ID]} found in Workspace settings`,
+            );
+            return;
+        }
+
+        const affinityConfigGlobalValue = affinityConfiguration?.globalValue;
+        if (affinityConfigGlobalValue && EXT_ID in affinityConfigGlobalValue) {
+            this.logger.debug(`Extension affinity value ${affinityConfigGlobalValue[EXT_ID]} found in User settings`);
+            return;
+        }
+
+        this.logger.debug("Extension affinity value not set in User and Workspace settings");
+
+        const defaultAffinity = 1;
+
+        const setAffinity = (value: number): void => {
+            this.logger.debug(`Setting extension affinity value to ${value} in User settings`);
+            extensionsConfiguration
+                .update("experimental.affinity", { ...affinityConfigGlobalValue, [EXT_ID]: value }, true)
+                .then(
+                    () => {
+                        this.logger.debug(`Successfull set extension affinity value to ${value} in User settings`);
+                    },
+                    (error) => {
+                        this.logger.error(`Error while setting experimental affinity. ${error}`);
+                    },
+                );
+        };
+
+        vscode.window
+            .showWarningMessage(
+                "No affinity assigned to vscode-neovim. It is recommended to assign affinity for major performance improvements. Would you like to set default affinity?",
```

Comment:
vscode messages support URLs. Would be very nice to link to the relevant documentation here.
```suggestion
                "No [affinity](https://github.com/vscode-neovim/vscode-neovim/issues/1051) assigned to vscode-neovim. It is recommended to assign affinity for major performance improvements. Would you like to set default affinity?",
```

Reply:
I have added a paragraph on affinity on the docs and added a link to it on the prompt. The discussions on #1051 don't quiet explain the feature.